### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by jediknight813** [ganelonhb/stable-diffusion_gd#1]
Can this work on android? There's a lot of cool applications that could be made using this, I'm surprised it hasn't taken off.

**Comment by ganelonhb** [comment]
Good question! I have not tried to run it on android, but as of right now Vulkan support is really borked. I have to figure out a way to integrate it with Godot's version of Vulkan. You may find success with the CPU backend, provided your game uses something like sdmini and the lcm weights.

It hasn't taken off because I have yet to do any evangelizing. It's not really fully ready in my eyes but I would like to get it there!

**Comment by jediknight813** [comment]
I'll try it out with the CPU backend, I'm thinking about making a VR application that runs on the meta quest 3, It has 8 gigs of ram plus a good cpu, with sd-turbo it could probably generate an image in 20 seconds.

Thanks for working on this, I hope it takes off, I think image generation/text generation in games is going to become more mainstream over the next few years.

Some people have been working on getting llama to run in Godot and it works pretty well:
https://github.com/nobodywho-ooo/nobodywho

**Comment by ganelonhb** [comment]
Thank you, in the coming months or so I should be updating this to the more recent version of stable-diffusion.cpp and I'll probably make it a bit more user friendly. With a newer version of GGML and stable-diffusion.cpp with flash attention merged in, it should run even better on mobile devices.

**Comment by wandbrandon** [comment]
Hey I'm looking at this repo for my implementation of exposing sd.cpp into Godot as well. I know that godot-llm exists which is relatively robust, but I haven't written any cpp since uni haha, let alone touching graphics backends. I do have a M3 Mac to test with if you need anything

**Comment by wandbrandon** [comment]
yeah problem is sd.cpp has it's own issues with Metal, Vulkan, and ggml_conv_2d's implementation. I saw this, which rewrites it using winograd. Looks interesting https://github.com/SealAILab/stable-diffusion-cpp?tab=readme-ov-file.
